pandas/core/generic.py

```diff
@@ -246,6 +248,23 @@ def attrs(self) -> Dict[Optional[Hashable], Any]:
     def attrs(self, value: Mapping[Optional[Hashable], Any]) -> None:
         self._attrs = dict(value)
 
+    @property
+    def allows_duplicate_labels(self) -> bool:
```

Comment:
I think better to make these private?

Reply:
This allows for setting after the NDFrame is created.

Alternatively (or in addition to) we could have a method that set the property to facilitate method chaining?

Reply:
Hmm OK. I wasn't expecting this to be a documented mutable property. Do you think that mutability is a long term requirement or is this just being add to aid working with the frame while it is experimental?

Reply:
I'd think long-term. I can imagine a workflow processing messy raw data like

```python
df = pd.read_csv("...", index_col="A")
# fixup duplicates in raws, messy data
df = df.groupby(level=0).first()

# Disallow duplicates going forward
df.allows_duplicate_labels = False
```

Reply:
Cool - I think that makes sense then